examples/Spec/Int.hs

```diff
@@ -0,0 +1,89 @@
+{-# LANGUAGE TypeFamilies #-}
+
+module Spec.Int (HasLogicalModel (..), IntProp(..),intGenTests,intPureTests,intPlutarchTests) where
+import Proper.HasLogicalModel
+import Proper.LogicalModel
+import Proper.HasParameterisedGenerator
+import Proper.HasPlutusTestRunner
+import Proper.HasPureTestRunner
+import SAT.MiniSat ( Formula (..) )
+import Hedgehog (forAll)
+import qualified Hedgehog.Gen as Gen
+import Hedgehog.Range (linear)
+import Data.Proxy (Proxy(..))
+import Test.Tasty (TestTree,testGroup)
+import Test.Tasty.Hedgehog (fromGroup)
+
+import Plutarch (compile)
+import Plutarch.Prelude
+
+data IntProp =
+      IsNegative
+    | IsPositive
+    | IsZero
+    | IsLarge
+    | IsSmall
+    | IsMaxBound
+    | IsMinBound
+    deriving stock (Eq,Ord,Enum,Show,Bounded)
+
+instance LogicalModel IntProp where
+  logic = ExactlyOne [Var IsNegative, Var IsPositive, Var IsZero]
+     :&&: ExactlyOne [Var IsLarge, Var IsSmall]
+     :&&: (Var IsZero :->: Var IsSmall)
+     :&&: (Var IsMaxBound :->: (Var IsLarge :&&: Var IsPositive))
+     :&&: (Var IsMinBound :->: (Var IsLarge :&&: Var IsNegative))
+
+instance HasLogicalModel Int IntProp where
+  satisfiesProperty i IsNegative = i < 0
+  satisfiesProperty i IsPositive = i > 0
+  satisfiesProperty i IsMaxBound = i == maxBound
+  satisfiesProperty i IsMinBound = i == minBound
+  satisfiesProperty i IsZero     = i == 0
+  satisfiesProperty i IsLarge    = i > 10 || i < -10
+  satisfiesProperty i IsSmall    = i <= 10 && i >= -10
+
+instance HasParameterisedGenerator Int IntProp where
+  parameterisedGenerator s = forAll $ do
+    i <- if IsZero `elem` s
+           then pure 0
+           else if IsSmall `elem` s
+                  then Gen.int (linear 1 10)
+                  else if IsMaxBound `elem` s
+                         then pure maxBound
+                         else Gen.int (linear 11 (maxBound -1))
+    if IsNegative `elem` s
+       then if IsMinBound `elem` s
+               then pure minBound
+               else pure (-i)
+       else pure i
```

Comment:
Now we define a parameterisedGenerator that can generate a model that satisfies a set of properties given a set of properties.